x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_rule_with_fallback.tsx

```diff
@@ -111,3 +125,27 @@ export const useRuleWithFallback = (ruleId: string): UseRuleWithFallback => {
     isExistingRule,
   };
 };
+
+/**
+ * Transforms an alertHit into a Rule
+ * @param data raw response containing single alert
+ */
+export const transformRuleFromAlertHit = (data: AlertSearchResponse): Rule | undefined => {
+  const hit = data?.hits.hits[0] as AlertHit | undefined;
+
+  // If pre 8.x alert, pull directly from alertHit
+  const rule = hit?._source.signal?.rule ?? hit?._source.kibana?.alert?.rule;
+
+  // If rule undefined, response likely flattened
+  if (rule == null) {
```

Comment:
Would be great if we cover that logic with tests. We already have unit tests for `useRuleWithFallback` in `use_rule_with_fallback.test.tsx`.

Reply:
Thanks @xcrzx -- I'll try to clean up the types a little and add some tests here as well 👍

Reply:
Coverage added in 9f5c5e9. Jest was a nuisance in overriding the default mock coming from `../alerts/__mocks__/api.ts`, but one of [these tips](https://stackoverflow.com/questions/48790927/how-to-change-mock-implementation-on-a-per-single-test-basis-jestjs) ended up working.